File: parkloyalty_main/lib/app/shared/widgets/activity_update_request.dart
// To parse this JSON data, do
//
//     final activityUpdateRequest = activityUpdateRequestFromJson(jsonString);

import 'dart:convert';

ActivityUpdateRequest activityUpdateRequestFromJson(String str) =>
    ActivityUpdateRequest.fromJson(json.decode(str));

String activityUpdateRequestToJson(ActivityUpdateRequest data) =>
    json.encode(data.toJson());

class ActivityUpdateRequest {
  String activityType;
  dynamic activityId;
  String clientTimestamp;
  String initiatorId;
  String initiatorRole;
  double? latitude;
  String logType;
  double? longitude;
  String siteId;
  String activityName;
  String shift;
  dynamic block;
  dynamic street;
  dynamic side;
  dynamic squad;
  dynamic deviceId;
  dynamic image1;
  dynamic image2;
  dynamic image3;
  bool isDisplay;
  String androidId;

  ActivityUpdateRequest({
    required this.activityType,
    required this.activityId,
    required this.clientTimestamp,
    required this.initiatorId,
    required this.initiatorRole,
    required this.latitude,
    required this.logType,
    required this.longitude,
    required this.siteId,
    required this.activityName,
    required this.shift,
    this.block,
    this.street,
    this.side,
    this.squad,
    this.deviceId,
    this.image1,
    this.image2,
    this.image3,
    required this.isDisplay,
    required this.androidId,
  });

  factory ActivityUpdateRequest.fromJson(Map<String, dynamic> json) =>
      ActivityUpdateRequest(
        activityType: json["activity_type"],
        activityId: json["activity_id"],
        clientTimestamp: json["client_timestamp"],
        initiatorId: json["initiator_id"],
        initiatorRole: json["initiator_role"],
        latitude: json["latitude"]?.toDouble(),
        logType: json["log_type"],
        longitude: json["longitude"]?.toDouble(),
        siteId: json["site_id"],
        activityName: json["activity_name"],
        shift: json["shift"],
        block: json["block"],
        street: json["street"],
        side: json["side"],
        squad: json["squad"],
        deviceId: json["device_id"],
        image1: json["image_1"],
        image2: json["image_2"],
        image3: json["image_3"],
        isDisplay: json["is_display"],
        androidId: json["android_id"],
      );

  Map<String, dynamic> toJson() => {
    "activity_type": activityType,
    "activity_id": activityId,
    "client_timestamp": clientTimestamp,
    "initiator_id": initiatorId,
    "initiator_role": initiatorRole,
    "latitude": latitude,
    "log_type": logType,
    "longitude": longitude,
    "site_id": siteId,
    "activity_name": activityName,
    "shift": shift,
    "block": block,
    "street": street,
    "side": side,
    "squad": squad,
    "device_id": deviceId,
    "image_1": image1,
    "image_2": image2,
    "image_3": image3,
    "is_display": isDisplay,
    "android_id": androidId,
  };
}

File: AndroidAppBeta-dev_target_34_code_base/app/src/main/java/com/parkloyalty/lpr/scan/ui/login/model/ActivityUpdateRequest.kt
package com.parkloyalty.lpr.scan.ui.login.model

import android.os.Parcelable
import androidx.annotation.Keep
import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonProperty
import kotlinx.parcelize.Parcelize

@Keep
@Parcelize
@JsonIgnoreProperties(ignoreUnknown = true, value = ["stability"])
data class ActivityUpdateRequest(
    @field:JsonProperty("activity_type")
    @get:JsonProperty("activity_type")
    var activityType: String? = null,
    @field:JsonProperty("activity_id")
    @get:JsonProperty("activity_id")
    var activity_id: String? = null,
    @field:JsonProperty("client_timestamp")
    @get:JsonProperty("client_timestamp")
    var clientTimestamp: String? = null,
    @field:JsonProperty("initiator_id")
    @get:JsonProperty("initiator_id")
    var initiatorId: String? = null,
    @field:JsonProperty("initiator_role")
    @get:JsonProperty("initiator_role")
    var initiatorRole: String? = null,
    @field:JsonProperty("latitude")
    @get:JsonProperty("latitude")
    var latitude: Double? = null,
    @field:JsonProperty("log_type")
    @get:JsonProperty("log_type")
    var logType: String? = null,
    @field:JsonProperty("longitude")
    @get:JsonProperty("longitude")
    var longitude: Double? = null,
    @field:JsonProperty("site_id")
    @get:JsonProperty("site_id")
    var siteId: String? = null,
    @field:JsonProperty("activity_name")
    @get:JsonProperty("activity_name")
    var activityName: String? = null,
    @field:JsonProperty("shift")
    @get:JsonProperty("shift")
    var mShift: String? = null,
    @field:JsonProperty("block")
    @get:JsonProperty("block")
    var mBlock: String? = null,
    @field:JsonProperty("street")
    @get:JsonProperty("street")
    var mStreet: String? = null,
    @field:JsonProperty("side")
    @get:JsonProperty("side")
    var mSide: String? = null,
    @field:JsonProperty("squad")
    @get:JsonProperty("squad")
    var mSquad: String? = null,
    @field:JsonProperty("device_id")
    @get:JsonProperty("device_id")
    var mDeviceId: String? = null,
    @field:JsonProperty("image_1")
    @get:JsonProperty("image_1")
    var image_1: String? = null,
    @field:JsonProperty("image_2")
    @get:JsonProperty("image_2")
    var image_2: String? = null,
    @field:JsonProperty("image_3")
    @get:JsonProperty("image_3")
    var image_3: String? = null,
    @field:JsonProperty("is_display")
    @get:JsonProperty("is_display")
    var isDisplay: Boolean? = null,
    @field:JsonProperty("android_id")
    @get:JsonProperty("android_id")
    var androidId: String? = ""
) : Parcelable
